SimTKcommon/include/SimTKcommon/internal/common.h

```diff
@@ -456,6 +456,7 @@ public:                                     \
     void invalidate(){clear();}                     \
     void clear(){ix=SimTK::InvalidIndex;}           \
     \
+    bool operator==(const NAME& other) const { return other == ix; } \
```

Comment:
I'd maybe consider something like:

```c++
bool operator==(const NAME& other) const { assert(isValidExtended() && isValidExtended(other)); return other.ix == ix; }
```

To match the other overloads

Reply:
Since `ix` is private I preferred to just reuse existing code without changing the interface of the class.

Reply:
Is `ix` being `private` an issue? The equality method here, `SimTK::NAME::operator==(const NAME& other) const`, is a member method of `NAME`, so it has full access to all of `NAME`'s private variables.

Reply:
Yea no sorry, Got confused by macros.
Then I agree with your suggestion. Fixed [here ](https://github.com/simbody/simbody/pull/748/commits/687ea74ca312f2b567e18d70f7c09d1abd5a1236), Thanks!